homeassistant/components/palazzetti/climate.py

```diff
@@ -0,0 +1,164 @@
+"""Support for Palazzetti climates."""
+
+from typing import Any, cast
+
+from palazzetti_sdk_local_api.exceptions import InvalidStateTransitionError
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ServiceValidationError
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    ACTION_NOT_UNAVAILABLE,
+    API_HW_VERSION,
+    API_NAME,
+    API_SW_VERSION,
+    AVAILABLE,
+    DOMAIN,
+    FAN_AUTO,
+    FAN_HIGH,
+    FAN_MODE,
+    FAN_MODES,
+    FAN_SILENT,
+    HEATING_STATUSES,
+    MAC,
+    MODE,
+    PALAZZETTI,
+    ROOM_TEMPERATURE,
+    TARGET_TEMPERATURE,
+)
+from .coordinator import PalazzettiDataUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Palazzetti climates based on a config entry."""
+    coordinator: PalazzettiDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    entities: list[PalazzettiClimateEntity] = []
+    await coordinator.async_config_entry_first_refresh()
```

Comment:
better to do that in `init` instead

Reply:
Not sure I get that. Can I await an async call in `init`? I'll look into it.

Reply:
Sorry, meant `__init__.py` in setup_entry

Reply:
Got it. Addressed in the new revision.